ge.c

```diff
@@ -13,6 +15,12 @@ static int ge_halted(struct ge *ge)
 
 int ge_init(struct ge *ge)
 {
+    ge->ge_console_socket = console_socket_init();
```

Comment:
I would prefer a middle layer, with peris that register init/on_cycle and deinit functions callbacks